client/src/connection/session.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright © 2022-2023, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { LogLine, OnLogFn, RunResult } from ".";
+
+export class BaseSession {
+  _onLogFn: OnLogFn;
+  public set onLogFn(value: (logs: LogLine[]) => void) {
+    this._onLogFn = value;
+  }
+}
+
+export interface Session {
+  onLogFn: (logs: LogLine[]) => void;
```

Comment:
We can define Session as a class with needed base implementation and abstract methods. A class can be used as an interface. It can avoid duplicated `onLogFn` declaration.

Reply:
A cleaner approach for sure. See Updated.